### PR TITLE
Debugging relativity on numba_montecarlo

### PR DESCRIPTION
**Pull request by arjunsavel**
## Description
**I'm working to fix the relativity functionality on the numba_montecarlo branch.** 
- Specific to relativity, I've corrected a distance calculation equation. 
- I've added missing `montecarlo_configuration` arguments to function calls
- I've removed a kwarg from the `MonteCarloConfiguration`, as [numba jitclass does not currently support kwargs](https://github.com/numba/numba/issues/4495).
- Added logging decorator capabilities
- Added debugging capabailities
- Added logging / debugging documentation
- Reverted the logic prior to the MonteCarloException; this now runs again, so long as `enable_full_relativity` is false.

## Motivation and Context
Full relativity currently isn't working on the numba_montecarlo branch, preventing it from being merged with master.

## How Has This Been Tested?
Advanced unit tests are currently failing due to. I can write tests after I address these failings to ensure that the relativity portion runs as expected. **No new bugs seem to be introduced.**

## Screenshots (if appropriate):
I can add screenshots if desired.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by harpolea** [comment]
I noticed that `@njit` has been converted to `@jit` in several places - is there a reason for this?

**Comment by arjunsavel** [comment]
@harpolea This is just for debugging reasons — in `nopython` mode, the `MonteCarloException` wasn't showing up as such, and I was just getting a `TypingError`. Once I figure out what's triggering the exception, I'm planning on changing the `@jit`s back to `@njit`s. Alternatively, I could figure out a better way to have exceptions shown in `nopython` mode...

File: tardis/montecarlo/montecarlo_numba/montecarlo_logger.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.DEBUG)
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(logging.DEBUG)
+    console_formatter = logging.Formatter(
+        '%(name)s - %(levelname)s - %(message)s')
+    console_handler.setFormatter(console_formatter)
+    logger.addHandler(console_handler)
+    return wrapper
```
Comment (wkerzendorf) [inline tardis/montecarlo/montecarlo_numba/montecarlo_logger.py:47]: I think add a newline

File: tardis/montecarlo/montecarlo_numba/r_packet.py [hunk 2]
```diff
@@ -89,15 +92,19 @@ def calculate_distance_line(r_packet, comov_nu, nu_line, time_explosion,
...
         nu_diff = 0.0
     if nu_diff <= 0:
-        print('nu difference is less than 0.0', nu_diff, comov_nu, nu, nu_line, time_explosion)
-        raise MonteCarloException('nu difference is less than 0.0')
+        print(f'nu difference is less than 0.0; current '
+                                  f'packet nu={r_packet.nu}, '\
+                                  f'packet seed={r_packet.index}, '\
+                                  f'nu line: {nu_line}',
+                                    f'shell id: {r_packet.current_shell_id}')
+        raise MonteCarloException()
```
Comment (wkerzendorf) [inline tardis/montecarlo/montecarlo_numba/r_packet.py:100]: add the string to the `MonteCarloException` and not `print`

**Pull request merged**